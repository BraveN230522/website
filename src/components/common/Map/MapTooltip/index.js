import React, { useCallback } from 'react';
import Router from 'next/router';
import moment from 'moment';

import {
  StyledTooltip,
  TooltipContentWrapper,
  Headline,
  VenueName,
  VenueAddress,
  MoreEvents,
  NavigationLink,
  SvgNavigation,
} from './styles';

const MapTooltip = ({
  id,
  dateTime,
  fromDate,
  toDate,
  headline,
  venueId,
  venueName,
  venueAddress,
  otherEventsCount,
  hideLink,
}) => {
  const query = {
    // default to today's date
    fromDate: moment(fromDate || new Date()).format('YYYY-MM-DD'),
    // default to null
    toDate: toDate && moment(new Date()).format('YYYY-MM-DD'),
  };

  const pushEventRoute = useCallback(() => {
    Router.push({
      pathname: `/event/${id}`,
    });
  }, [id]);

  const pushVenueRoute = useCallback(() => {
    Router.push({
      pathname: `/venue/${venueId}`,
      query,
    });
  }, [id]);

  return (
    <StyledTooltip
      id={id}
      aria-haspopup="true"
      type="light"
      effect="solid"
      delayHide={500}
      delayShow={500}
      delayUpdate={500}
    >
      <TooltipContentWrapper>
        <Headline onClick={pushEventRoute}>{headline}</Headline>
        <VenueName>{venueName}</VenueName>
        <VenueAddress>{venueAddress}</VenueAddress>
        {Boolean(dateTime) && <VenueAddress>{moment(Number(dateTime)).format('DD MMM YYYY')}</VenueAddress>}

        {otherEventsCount > 0 && (
          <MoreEvents onClick={pushVenueRoute}>
            See {otherEventsCount} more {otherEventsCount === 1 ? 'event' : 'events'}
          </MoreEvents>
        )}
        {!hideLink && (
          <NavigationLink onClick={pushEventRoute}>
            <SvgNavigation />
          </NavigationLink>
        )}
      </TooltipContentWrapper>
    </StyledTooltip>
  );
};

export default MapTooltip;
