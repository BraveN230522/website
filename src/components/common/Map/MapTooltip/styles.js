import React from 'react';
import styled from '@emotion/styled';
import ReactTooltip from 'react-tooltip';

const SvgNavigation = () => (
  <svg width="16" height="28" viewBox="0 0 16 28" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M0 2.30739L12.3001 13.999L0.000981601 25.6902L1.79731 27.5L16 13.999L1.79731 0.5L0 2.30739Z"
      fill="#2A98FF"
    />
  </svg>
);

const TOOLTIP_ARROW_SIZE = '20px';

const StyledTooltip = styled(ReactTooltip)`
  /* Each '&' creates a new random class name. */
  /* With '&&.something' e create a selector with 2 new random class names. */
  /* This means that '&&.something' will create a selector with 2 new random class names */
  /* Therefore, we will have a CSS selector with a higher specificity level without using '!important' in this case */
  &&.__react_component_tooltip {
    background-color: ${({ theme }) => theme.colors.white};
    padding: 0;
    border-radius: 9px;
    box-shadow: 0px 2px 27px rgba(0, 0, 0, 0.4);

    &:hover {
      cursor: default;
    }

    &.place-top {
      margin-top: -${TOOLTIP_ARROW_SIZE};

      &::before {
        border-top: ${TOOLTIP_ARROW_SIZE} solid transparent;
      }

      &::after {
        border-left: ${TOOLTIP_ARROW_SIZE} solid transparent;
        border-right: ${TOOLTIP_ARROW_SIZE} solid transparent;
        bottom: -${TOOLTIP_ARROW_SIZE};
        left: calc(50% - ${TOOLTIP_ARROW_SIZE});
        margin-left: 0;
        border-top-color: ${({ theme }) => theme.colors.white};
        border-top-style: solid;
        border-top-width: ${TOOLTIP_ARROW_SIZE};
      }
    }

    &.place-bottom {
      margin-top: ${TOOLTIP_ARROW_SIZE};

      &::before {
        border-bottom: ${TOOLTIP_ARROW_SIZE} solid transparent;
      }

      &::after {
        border-left: ${TOOLTIP_ARROW_SIZE} solid transparent;
        border-right: ${TOOLTIP_ARROW_SIZE} solid transparent;
        top: -${TOOLTIP_ARROW_SIZE};
        left: calc(50% - ${TOOLTIP_ARROW_SIZE});
        margin-left: 0;
        border-bottom-color: ${({ theme }) => theme.colors.white};
        border-bottom-style: solid;
        border-bottom-width: ${TOOLTIP_ARROW_SIZE};
      }
    }

    &.place-left {
      margin-left: -${TOOLTIP_ARROW_SIZE};

      &::before {
        border-left: ${TOOLTIP_ARROW_SIZE} solid transparent;
      }

      &::after {
        border-top: ${TOOLTIP_ARROW_SIZE} solid transparent;
        border-bottom: ${TOOLTIP_ARROW_SIZE} solid transparent;
        right: -${TOOLTIP_ARROW_SIZE};
        top: calc(50% - ${TOOLTIP_ARROW_SIZE});
        margin-top: 0;
        border-left-color: ${({ theme }) => theme.colors.white};
        border-left-style: solid;
        border-left-width: ${TOOLTIP_ARROW_SIZE};
      }
    }

    &.place-right {
      margin-left: ${TOOLTIP_ARROW_SIZE};

      &::before {
        border-right: ${TOOLTIP_ARROW_SIZE} solid transparent;
        border-right: ${TOOLTIP_ARROW_SIZE} solid white;
      }

      &::after {
        border-top: ${TOOLTIP_ARROW_SIZE} solid transparent;
        border-bottom: ${TOOLTIP_ARROW_SIZE} solid transparent;
        left: -${TOOLTIP_ARROW_SIZE};
        top: calc(50% - ${TOOLTIP_ARROW_SIZE});
        margin-top: 0;
        border-right-color: ${({ theme }) => theme.colors.white};
        border-right-style: solid;
        border-right-width: ${TOOLTIP_ARROW_SIZE};
      }
    }
  }
`;

const TooltipContentWrapper = styled.div`
  position: relative;
  padding: 24px;
  border-radius: 9px;

  display: flex;
  flex-direction: column;
  width: 339px;
  background-color: ${({ theme }) => theme.colors.white};
`;

const Headline = styled.a`
  opacity: 0.8;
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-style: normal;
  font-size: 17px;
  font-weight: bold;
  letter-spacing: -0.12px;
  color: ${({ theme }) => theme.colors.swiamBlue};

  &:hover {
    text-decoration: none;
    cursor: pointer;
  }
`;

const VenueName = styled.span`
  opacity: 0.8;
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-style: normal;
  font-size: 14px;
  letter-spacing: -0.1px;
  color: ${({ theme }) => theme.colors.black};
`;

const VenueAddress = styled.p`
  margin: 0;
  padding: 0;
  opacity: 0.8;
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-style: normal;
  font-size: 12.3px;
  letter-spacing: -0.1px;
  color: ${({ theme }) => theme.colors.lightTooltip};
`;

const MoreEvents = styled.a`
  margin-top: 16px;
  opacity: 0.8;
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-size: 15px;
  font-weight: normal;
  font-style: normal;
  letter-spacing: -0.1px;
  color: ${({ theme }) => theme.colors.swiamBlue};

  &:hover {
    text-decoration: none;
    cursor: pointer;
  }
`;

const NavigationLink = styled.a`
  position: absolute;
  right: 16px;
  top: 45.5px;
  &:hover {
    cursor: pointer;
  }
`;

export {
  StyledTooltip,
  TooltipContentWrapper,
  Headline,
  VenueName,
  VenueAddress,
  MoreEvents,
  NavigationLink,
  SvgNavigation,
};
