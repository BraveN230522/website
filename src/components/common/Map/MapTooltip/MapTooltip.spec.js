import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import MapTooltip from './index';

jest.mock('crypto', () => ({
  randomBytes: num => new Array(num).fill(0),
}));

describe('<MapTooltip />', () => {
  it('Renders the MapTooltip', () => {
    const { container } = render(
      <MapTooltip id="id-123" headline="Great event" venueName="Online" venueAddress="Your web browser" />,
      {
        wrapper: withGlobals,
      }
    );

    expect(container).toMatchSnapshot();
  });
});
