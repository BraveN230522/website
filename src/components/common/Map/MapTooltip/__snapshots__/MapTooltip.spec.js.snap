// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapTooltip /> Renders the MapTooltip 1`] = `
.emotion-8 {
  position: relative;
  padding: 24px;
  border-radius: 9px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 339px;
  background-color: #ffffff;
}

.emotion-0 {
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 17px;
  font-weight: bold;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  color: #2891ff;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-2 {
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 14px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  color: #000000;
}

.emotion-4 {
  margin: 0;
  padding: 0;
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 12.3px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  color: #6c6b6b;
}

.emotion-6 {
  position: absolute;
  right: 16px;
  top: 45.5px;
}

.emotion-6:hover {
  cursor: pointer;
}

<div>
  <div
    aria-haspopup="true"
    class="__react_component_tooltip t00000000-0000-4000-8000-000000000000 place-top type-dark allow_hover"
    data-id="tooltip"
    id="id-123"
  >
    <style>
      
  	.t00000000-0000-4000-8000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-4000-8000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
    </style>
    <div
      class="emotion-8 emotion-9"
    >
      <a
        class="emotion-0 emotion-1"
      >
        Great event
      </a>
      <span
        class="emotion-2 emotion-3"
      >
        Online
      </span>
      <p
        class="emotion-4 emotion-5"
      >
        Your web browser
      </p>
      <a
        class="emotion-6 emotion-7"
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 16 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 2.30739L12.3001 13.999L0.000981601 25.6902L1.79731 27.5L16 13.999L1.79731 0.5L0 2.30739Z"
            fill="#2A98FF"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
